gamemodes/terrortown/gamemode/server/sv_weaponry.lua

```diff
@@ -652,6 +656,13 @@ function WEPS.DropNotifiedWeapon(ply, wep, deathDrop, keepSelection)
 	ply:DropWeapon(wep)
 
 	wep:PhysWake()
+
+	-- Unset this, because we can't use it after this point.
+	wep.IsDroppedBecauseDeath = nil
+
+	if deathDrop and wep.overrideDropOnDeath == DROP_ON_DEATH_TYPE_DENY then
+		wep:Remove()
+	end
```

Comment:
Can we move this up a bit, to avoid calling dropWeapon at all, if it is removed anyways?

Reply:
it could be possible, i just know some guns such as the defib use OnDrop to cancel revives, or to clean up placement ghosts -- i don't know if i can always count on them to do that during Destroy as well

Reply:
Puh honestly i think this is safe to assume. But ultimately i think its much better not to drop the weapon when deleting it instantly afterwards. This could cause weird behavior. 

What is your opinion @Histalek ?

Reply:
I agree with saibotk on this one. We should simply do what we intend to do and that is simply removing the weapon/equipment.

If the equipment does not correctly clean up after itself in case of a removal, i'd consider that a bug of the equipment. We should not need to work around that.

Reply:
makes sense to me, glad to have some valuable perspective on it